src/Codeception/Lib/Driver/Sqlite.php

```diff
@@ -22,19 +21,24 @@ public function __construct($dsn, $user, $password)
         parent::__construct($this->dsn, $user, $password);
     }
 
-    public function cleanup()
+    public function __destruct()
     {
-        $this->dbh = null;
         file_put_contents($this->filename, '');
-        $this->dbh = self::connect($this->dsn, $this->user, $this->password);
+    }
+
+    public function cleanup()
+    {
+        $this->dbh->exec('PRAGMA foreign_keys = OFF;');
+        $this->dbh->exec('PRAGMA writable_schema = 1;');
+        $this->dbh->exec('DELETE FROM sqlite_master WHERE type IN (\'table\', \'index\', \'trigger\');');
```

Comment:
@DavertMik also, the "cleanup" was emptying the file `file_put_contents($this->filename, '');` and then force a reconnection to avoid a "schema changed" error. That was coupled with the Db module implementation, so after I change it to make "more sense" this coupling broke. All drivers "SQL the cleanup" all the way...